### DISCUSSION TITLE
QOA - Quite OK Audio. Fast, 0.45ms delay, ~300kbps

### DISCUSSION THREAD

**Discussion by 38github** [jamulussoftware/jamulus#3023] [Ideas]
Has anyone seen this codec? Would it be useable for something like Jamulus?

https://github.com/phoboslab/qoa

**Comment by ann0see** [comment]
It sounds very beta from the README file. I'd rather wait and improve issues we face with Opus. 
You can of course talk to the author and make him aware of Jamulus/HPS Jam/...

**Comment by 38github** [comment]
@phoboslab Do you see your codec as something that can be used as an improvement in Jamulus? Thanks in advance!

**Comment by phoboslab** [comment]
Opus seems like the logical choice if you're not CPU bound. What are the issues you're facing?

I have not yet implemented QOA in any streaming application, so I'm not sure how well it would work. Since QOA doesn't have any built-in error correction, you'd either need a reliable transport (i.e. TCP), or roll your own wrapper to reorder and validate packets. The theoretical 0.45ms delay is accurate, but likely dwarfed by the necessary buffering on the receiving end to account for network jitter.

Using QOA for this would certainly be an interesting project, but Opus is battle tested and probably very hard to compete with.

**Reply by 38github** [reply]
Thank you, Phoboslab, for your very informative answer. I am interested in the possible very low latency and that it sounds very good at low bitrate. The only artifacts I have heard in a few samples are high frequency noise and I do not find it annoying at all. If I didn't know the input audio I would have never really known if that small amount of noise was an artifact or actually in the audio to begin with.
